libcontainer/cgroups/ebpf/ebpf.go

```diff
@@ -1,29 +1,93 @@
 package ebpf
 
 import (
+	"runtime"
+	"unsafe"
+
 	"github.com/cilium/ebpf"
 	"github.com/cilium/ebpf/asm"
 	"github.com/cilium/ebpf/link"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"golang.org/x/sys/unix"
 )
 
+func nilCloser() error {
+	return nil
+}
+
+func findAttachedCgroupDeviceFilters(dirFd int) ([]*ebpf.Program, error) {
+	type bpfAttrQuery struct {
+		TargetFd    uint32
+		AttachType  uint32
+		QueryType   uint32
+		AttachFlags uint32
+		ProgIds     uint64 // __aligned_u64
+		ProgCnt     uint32
+	}
+
+	// Currently you can only have 64 eBPF programs attached to a cgroup.
+	size := 64
+	allowRetry := true
+	for allowRetry {
```

Comment:
It seems that `allowRetry` is redundant; you can just do

```go
for {
	...
	if errno != 0 {
		// On ENOSPC we get the correct number of programs.
		if errno == unix.ENOSPC {
			size = int(query.ProgCnt)
			continue
		}
		return nil, fmt.Errorf("bpf_prog_query: %w", errno)
	}
	...
	return ...
}
```

Reply:
I don't want to loop endlessly ... but you're quite right that the `allowRetry` logic was wrong (oops). I'll fix it up.